compiler/lib/Conversion/LcclToByre/LcclToByre.cpp

```diff
@@ -31,6 +31,28 @@ using namespace llvm;
 
 namespace {
 
+template <typename T>
+ArrayAttr getByreMemoryEffectForLcclOps(T op, PatternRewriter &rewriter) {
+  SmallVector<Attribute> memoryEffects;
+  if (MemoryEffectOpInterface iface =
+          dyn_cast<MemoryEffectOpInterface>(op.getOperation())) {
+    for (auto memrefOp : op.getOperands()) {
+      if (iface.getEffectOnValue<MemoryEffects::Write>(memrefOp)) {
```

Comment:
seems it doesn't handle effect `Write & Read`.

Reply:
I have made some revisions. Could you please review the code once more?